synchronizer/l1_rollup_info_producer.go

```diff
@@ -195,11 +217,17 @@ func (l *l1RollupInfoProducer) verify() error {
 	return l.syncStatus.verify()
 }
 
-func (l *l1RollupInfoProducer) initialize() error {
+func (l *l1RollupInfoProducer) initialize(ctx context.Context) error {
+	log.Debug("producer: initialize")
 	err := l.verify()
 	if err != nil {
 		return err
 	}
+	if l.ctxParent != ctx || l.ctx != nil || l.cancelCtx == nil || (l.cancelCtx != nil && l.ctx.Err() != nil) {
+		log.Debug("producer: start called and need to create a new context")
+		l.ctxParent = ctx
+		l.ctx, l.cancelCtx = context.WithCancel(l.ctxParent)
+	}
```

Comment:
```suggestion
	l.ctxParent = ctx
	if l.ctx != nil || l.cancelCtx == nil || (l.cancelCtx != nil && l.ctx.Err() != nil) {
		log.Debug("producer: start called and need to create a new context")
		l.ctx, l.cancelCtx = context.WithCancel(l.ctxParent)
	}
```

Reply:
I think that is not same: if `ctxParent` and incomming `ctx` we don't need to recreate contexts